src/main/java/dev/blachut/svelte/lang/codeInsight/SvelteReferenceContributor.kt

```diff
@@ -0,0 +1,70 @@
+package dev.blachut.svelte.lang.codeInsight
+
+import com.intellij.lang.javascript.psi.*
+import com.intellij.lang.javascript.psi.impl.JSReferenceExpressionImpl
+import com.intellij.lang.javascript.psi.resolve.JSReferenceExpressionResolver
+import com.intellij.lang.javascript.psi.resolve.JSResolveResult
+import com.intellij.openapi.util.TextRange
+import com.intellij.patterns.PlatformPatterns
+import com.intellij.psi.*
+import com.intellij.psi.filters.ElementFilter
+import com.intellij.psi.filters.position.FilterPattern
+import com.intellij.psi.impl.source.resolve.ResolveCache
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.util.ProcessingContext
+
+object DeclarationFinder {
+    fun findDeclaration(element: JSReferenceExpressionImpl?): Boolean {
+        element ?: return false
+        val cachedResults = ResolveCache.getInstance(element.project).resolveWithCaching(element, JSReferenceExpressionResolver(element, false), true, false)
+        return cachedResults.find {
+            it.element is JSVariable
+        } != null
+    }
+}
+
+class SvelteReferenceContributor : PsiReferenceContributor() {
+    override fun registerReferenceProviders(registrar: PsiReferenceRegistrar) {
+        val pattern = PlatformPatterns.psiElement(JSReferenceExpression::class.java).and(FilterPattern(object : ElementFilter {
+            override fun isAcceptable(element: Any, context: PsiElement?): Boolean {
+                if (element is JSReferenceExpressionImpl) {
+                    if (DeclarationFinder.findDeclaration(element)) {
+                        return false
+                    }
+                    return PsiTreeUtil.findFirstParent(element) { parent -> parent is JSLabeledStatement } == null
```

Comment:
I think this line prevents chained reactive declarations working.

<img width="458" alt="Screenshot 2019-08-07 at 14 59 28" src="https://user-images.githubusercontent.com/615013/62624804-2dcece80-b924-11e9-96b6-3d94914d0aaa.png">

Reply:
You mean `HighlightingVisitor`?

Reply:
I meant like `JSElementVisitor`

Reply:
I'm talking about this method to be exact
https://github.com/tomblachut/svelte-intellij/blob/1f3643ae8310ff27a1872da00dda2da354139dc2/src/main/java/dev/blachut/svelte/lang/codeInsight/SvelteReferenceContributor.kt#L52
Do you think it's efficient like this? Would a `JSElementVisitor` be better in this case?

Reply:
In the end visitor performs traversal so IDK similarly to find* methods, we would need to measure it, anyway I think it's okay for now. I'm gonna check your code again and merge it if nothing more turns up. Then I'm gonna try to tackle $ 
references.

Reply:
I just figured out that `PsiTreeUtil.findChildrenOfType` uses a recursive element visitor under the hood. So it's the same thing after all.